### PR TITLE
fix: handle unknown selector error

### PR DESCRIPTION
**Pull request by Wodann**
Fixes https://github.com/NomicFoundation/edr/issues/281

### CODE REVIEW

File: crates/edr_provider/src/error.rs [hunk 1]
```diff
@@ -399,10 +399,13 @@ fn revert_error(output: &Bytes) -> String {
...
         Err(decode_error) => match decode_error {
-            alloy_sol_types::Error::TypeCheckFail { .. } => {
+            alloy_sol_types::Error::TypeCheckFail { .. }
+            | alloy_sol_types::Error::UnknownSelector { .. } => {
                 format!("VM Exception while processing transaction: reverted with an unrecognized custom error (return data: 0x{})", hex::encode(output))
             }
-            _ => unreachable!("Since we are not validating, no other error can occur"),
+            _ => unreachable!(
+                "Since we are not validating, no other error can occur: {decode_error:?}"
+            ),
```
Comment (Wodann) [inline crates/edr_provider/src/error.rs:408]: @agostbiro you're more familiar with this code. Can we logically rule out that none of the other error variants can occur?

**Review by agostbiro — CHANGES_REQUESTED** [review]
Thanks for the quick fixes @Wodann ! I left some comments

File: crates/edr_provider/src/error.rs [hunk 2]
```diff
@@ -399,10 +399,13 @@ fn revert_error(output: &Bytes) -> String {
             }
         }
         Err(decode_error) => match decode_error {
-            alloy_sol_types::Error::TypeCheckFail { .. } => {
+            alloy_sol_types::Error::TypeCheckFail { .. }
+            | alloy_sol_types::Error::UnknownSelector { .. } => {
```
[the remark this answers is not in the document]
Comment (Wodann) [reply crates/edr_provider/src/error.rs:403]: I tested the test with the issue you mentioned and it did, yes.

I can double check

[the remark this answers is not in the document]
Comment (Wodann) [reply crates/edr_provider/src/error.rs:403]: You're right. It does not fix [#283](https://github.com/NomicFoundation/edr/issues/283). Aplogies

File: crates/edr_provider/src/error.rs (hunk 1 above)
Comment (Wodann) [reply crates/edr_provider/src/error.rs]: Addressed in https://github.com/NomicFoundation/hardhat/pull/4811/commits/3e86c0d9be3bcf6c1ba195134449e6c39f01e3b2

**Review by agostbiro — APPROVED** [review]

**Pull request merged**